app/domain/repos/conjur_ca.rb

```diff
@@ -14,12 +14,28 @@ def self.create(resource_id)
     #
     def self.ca(resource_id)
       ca_info = ::Conjur::CaInfo.new(resource_id)
-      stored_cert = Resource[ca_info.cert_id].last_secret.value
-      stored_key = Resource[ca_info.key_id].last_secret.value
+      stored_cert, stored_key = fetch_ca_secrets(ca_info, resource_id)
       ca_cert = OpenSSL::X509::Certificate.new(stored_cert)
       ca_key = OpenSSL::PKey::RSA.new(stored_key)
       ::Util::OpenSsl::CA.new(ca_cert, ca_key)
     end
 
+    class << self
+      private
+
+      def self.fetch_secret
+        @fetch_secret ||= ::Conjur::FetchRequiredSecrets.new
+      end
+
+      def self.fetch_ca_secrets(ca_info, resource_id)
+        begin
+          fetch_secret.(resource_ids: [ca_info.cert_id, ca_info.key_id])
+        rescue Errors::Conjur::RequiredResourceMissing
+          raise Errors::Authentication::AuthnK8s::CaResourceMissing, resource_id
```

Comment:
what is `resource_id` in this context? what will be printed to the log?

Reply:
Also - which log message will be the most helpful for the user?

Reply:
which message will give them the best pin-pointed action item?

Reply:
Refactor exceptions messages